### PR TITLE
rustdoc: display rustc_private APIs as "Internal"

### PR DESCRIPTION
**Pull request by euclio**
This PR updates the display of `rustc_private` APIs to be "Internal" instead of "Experimental", and changes the colors appropriately. It also updates the copy of the `rustc_private` feature to sound more informative and less like a compiler suggestion.

The PR additionally contains a significant refactor of the `short_stability` function to remove duplication and fix a few rendering bugs due to extra or missing spaces.

Before:

screen shot 2018-12-14 at 11 45 28 am

After:

screen shot 2018-12-14 at 11 45 11 am

### CODE REVIEW

File: src/librustdoc/html/static/themes/dark.css [hunk 1]
```diff
@@ -199,6 +203,7 @@ a.test-arrow {
 }
 
 .stab.unstable { background: #FFF5D6; border-color: #FFC600; color: #404040; }
+.stab.internal { background: #FF9A91; border-color: #B71C1C; color: #404040; }
```
[the remark this answers is not in the document]
Comment (euclio) [reply src/librustdoc/html/static/themes/dark.css]: Yeah, that color looks better to me too. Fixed.

**Review by QuietMisdreavus — APPROVED** [review]
Thanks so much! I like the refactors you did, it makes the code easier to read.

**Pull request merged**